{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "615f12a3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-20T20:59:28.686269Z",
     "start_time": "2021-09-20T20:59:19.015275Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d7231c5b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-20T20:59:28.734314Z",
     "start_time": "2021-09-20T20:59:28.691325Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = ['22241','22334','22243','22244','22245','22246','22247',\n",
    " '22248','22249','22250','22251','22252','22253','22254','22255']\n",
    "len(ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "055c5f8d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-20T20:59:28.829590Z",
     "start_time": "2021-09-20T20:59:28.740291Z"
    },
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.2\n",
    "##source=VarDict_v1.8.2\n",
    "##INFO=<ID=SAMPLE,Number=1,Type=String,Description=\"Sample name (with whitespace translated to underscores)\">\n",
    "##INFO=<ID=TYPE,Number=1,Type=String,Description=\"Variant Type: SNV Insertion Deletion Complex\">\n",
    "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Total Depth\">\n",
    "##INFO=<ID=END,Number=1,Type=Integer,Description=\"Chr End Position\">\n",
    "##INFO=<ID=VD,Number=1,Type=Integer,Description=\"Variant Depth\">\n",
    "##INFO=<ID=AF,Number=A,Type=Float,Description=\"Allele Frequency\">\n",
    "##INFO=<ID=BIAS,Number=1,Type=String,Description=\"Strand Bias Info\">\n",
    "##INFO=<ID=REFBIAS,Number=1,Type=String,Description=\"Reference depth by strand\">\n",
    "##INFO=<ID=VARBIAS,Number=1,Type=String,Description=\"Variant depth by strand\">\n",
    "##INFO=<ID=PMEAN,Number=1,Type=Float,Description=\"Mean position in reads\">\n",
    "##INFO=<ID=PSTD,Number=1,Type=Float,Description=\"Position STD in reads\">\n",
    "##INFO=<ID=QUAL,Number=1,Type=Float,Description=\"Mean quality score in reads\">\n",
    "##INFO=<ID=QSTD,Number=1,Type=Float,Description=\"Quality score STD in reads\">\n",
    "##INFO=<ID=SBF,Number=1,Type=Float,Description=\"Strand Bias Fisher p-value\">\n",
    "##INFO=<ID=ODDRATIO,Number=1,Type=Float,Description=\"Strand Bias Odds ratio\">\n",
    "##INFO=<ID=MQ,Number=1,Type=Float,Description=\"Mean Mapping Quality\">\n",
    "##INFO=<ID=SN,Number=1,Type=Float,Description=\"Signal to noise\">\n",
    "##INFO=<ID=HIAF,Number=1,Type=Float,Description=\"Allele frequency using only high quality bases\">\n",
    "##INFO=<ID=ADJAF,Number=1,Type=Float,Description=\"Adjusted AF for indels due to local realignment\">\n",
    "##INFO=<ID=SHIFT3,Number=1,Type=Integer,Description=\"No. of bases to be shifted to 3 prime for deletions due to alternative alignment\">\n",
    "##INFO=<ID=MSI,Number=1,Type=Float,Description=\"MicroSatellite. > 1 indicates MSI\">\n",
    "##INFO=<ID=MSILEN,Number=1,Type=Float,Description=\"MicroSatellite unit length in bp\">\n",
    "##INFO=<ID=NM,Number=1,Type=Float,Description=\"Mean mismatches in reads\">\n",
    "##INFO=<ID=LSEQ,Number=1,Type=String,Description=\"5' flanking seq\">\n",
    "##INFO=<ID=RSEQ,Number=1,Type=String,Description=\"3' flanking seq\">\n",
    "##INFO=<ID=GDAMP,Number=1,Type=Integer,Description=\"No. of amplicons supporting variant\">\n",
    "##INFO=<ID=TLAMP,Number=1,Type=Integer,Description=\"Total of amplicons covering variant\">\n",
    "##INFO=<ID=NCAMP,Number=1,Type=Integer,Description=\"No. of amplicons don't work\">\n",
    "##INFO=<ID=AMPFLAG,Number=1,Type=Integer,Description=\"Top variant in amplicons don't match\">\n",
    "##INFO=<ID=HICNT,Number=1,Type=Integer,Description=\"High quality variant reads\">\n",
    "##INFO=<ID=HICOV,Number=1,Type=Integer,Description=\"High quality total reads\">\n",
    "##INFO=<ID=SPLITREAD,Number=1,Type=Integer,Description=\"No. of split reads supporting SV\">\n",
    "##INFO=<ID=SPANPAIR,Number=1,Type=Integer,Description=\"No. of pairs supporting SV\">\n",
    "##INFO=<ID=SVTYPE,Number=1,Type=String,Description=\"SV type: INV DUP DEL INS FUS\">\n",
    "##INFO=<ID=SVLEN,Number=1,Type=Integer,Description=\"The length of SV in bp\">\n",
    "##INFO=<ID=DUPRATE,Number=1,Type=Float,Description=\"Duplication rate in fraction\">\n",
    "##FILTER=<ID=q22.5,Description=\"Mean Base Quality Below 22.5\">\n",
    "##FILTER=<ID=Q10,Description=\"Mean Mapping Quality Below 10\">\n",
    "##FILTER=<ID=p8,Description=\"Mean Position in Reads Less than 8\">\n",
    "##FILTER=<ID=SN1.5,Description=\"Signal to Noise Less than 1.5\">\n",
    "##FILTER=<ID=Bias,Description=\"Strand Bias\">\n",
    "##FILTER=<ID=pSTD,Description=\"Position in Reads has STD of 0\">\n",
    "##FILTER=<ID=d3,Description=\"Total Depth < 3\">\n",
    "##FILTER=<ID=v2,Description=\"Var Depth < 2\">\n",
    "##FILTER=<ID=f0.01,Description=\"Allele frequency < 0.01\">\n",
    "##FILTER=<ID=MSI12,Description=\"Variant in MSI region with 12 non-monomer MSI or 13 monomer MSI\">\n",
    "##FILTER=<ID=NM5.25,Description=\"Mean mismatches in reads >= 5.25, thus likely false positive\">\n",
    "##FILTER=<ID=InGap,Description=\"The variant is in the deletion gap, thus likely false positive\">\n",
    "##FILTER=<ID=InIns,Description=\"The variant is adjacent to an insertion variant\">\n",
    "##FILTER=<ID=Cluster0bp,Description=\"Two variants are within 0 bp\">\n",
    "##FILTER=<ID=LongMSI,Description=\"The somatic variant is flanked by long A/T (>=14)\">\n",
    "##FILTER=<ID=AMPBIAS,Description=\"Indicate the variant has amplicon bias.\">\n",
    "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
    "##FORMAT=<ID=DP,Number=1,Type=Integer,Description=\"Total Depth\">\n",
    "##FORMAT=<ID=VD,Number=1,Type=Integer,Description=\"Variant Depth\">\n",
    "##FORMAT=<ID=AD,Number=R,Type=Integer,Description=\"Allelic depths for the ref and alt alleles in the order listed\">\n",
    "##FORMAT=<ID=AF,Number=A,Type=Float,Description=\"Allele Frequency\">\n",
    "##FORMAT=<ID=RD,Number=2,Type=Integer,Description=\"Reference forward, reverse reads\">\n",
    "##FORMAT=<ID=ALD,Number=2,Type=Integer,Description=\"Variant forward, reverse reads\">\n",
    "##INFO=<ID=ANNOVAR_DATE,Number=1,Type=String,Description=\"Flag the start of ANNOVAR annotation for one alternative allele\">\n",
    "##INFO=<ID=Func.refGene,Number=.,Type=String,Description=\"Func.refGene annotation provided by ANNOVAR\">\n",
    "##INFO=<ID=Gene.refGene,Number=.,Type=String,Description=\"Gene.refGene annotation provided by ANNOVAR\">\n",
    "##INFO=<ID=GeneDetail.refGene,Number=.,Type=String,Description=\"GeneDetail.refGene annotation provided by ANNOVAR\">\n",
    "##INFO=<ID=ExonicFunc.refGene,Number=.,Type=String,Description=\"ExonicFunc.refGene annotation provided by ANNOVAR\">\n",
    "##INFO=<ID=AAChange.refGene,Number=.,Type=String,Description=\"AAChange.refGene annotation provided by ANNOVAR\">\n",
    "##INFO=<ID=ALLELE_END,Number=0,Type=Flag,Description=\"Flag the end of ANNOVAR annotation for one alternative allele\">\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f1dd2210",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-20T21:06:56.677809Z",
     "start_time": "2021-09-20T21:06:56.634052Z"
    }
   },
   "outputs": [],
   "source": [
    "for x in range(3):\n",
    "    for y in range(4):\n",
    "        \n",
    "        parent = ids[x*5]\n",
    "        variant = ids[x*5 + y+1]\n",
    "        \n",
    "        # file_name = \"results/VE_\" + parent + \"_VE_\" + variant + \".xlsx\"\n",
    "        file_name = \"results/test.xlsx\"\n",
    "        \n",
    "        pv_df = pd.read_excel(file_name)#, sheet_name=\"p-v \" + variant)\n",
    "        pv_df = pv_df[:10]\n",
    "        # vp_df = pd.read_excel(file_name, sheet_name=\"v-p \" + variant)\n",
    "\n",
    "        pv_vcf = parent + \"_\" + variant + \"_p-v.vcf\"\n",
    "        with open(pv_vcf, 'w') as vcf:\n",
    "            vcf.write(header)\n",
    "        \n",
    "        pv_df.to_csv(pv_vcf, sep=\"\\t\", mode='a', index=False)\n",
    "        \n",
    "        break\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a0abb43a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-20T21:24:55.005532Z",
     "start_time": "2021-09-20T21:24:54.997563Z"
    }
   },
   "outputs": [],
   "source": [
    "# sub_name must be p-v or v-p\n",
    "def create_vcf(sub_name):\n",
    "    df = pd.read_excel(file_name, sheet_name=sub_name + \" \" + variant)\n",
    "    vcf_file = parent + \"_\" + variant + \"_\" + sub_name + \".vcf\"\n",
    "    with open(vcf_file, 'w') as vcf:\n",
    "        vcf.write(header)\n",
    "    df.to_csv(vcf, sep=\"\\t\", mode='a', index=False)\n",
    "    return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8dbd1de7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-20T21:16:33.218165Z",
     "start_time": "2021-09-20T21:16:17.544182Z"
    }
   },
   "outputs": [],
   "source": [
    "for x in range(3):\n",
    "    for y in range(4):\n",
    "        \n",
    "        parent = ids[x*5]\n",
    "        variant = ids[x*5 + y+1]\n",
    "        \n",
    "        file_name = \"results/VE_\" + parent + \"_VE_\" + variant + \".xlsx\"\n",
    "\n",
    "        pv_df = pd.read_excel(file_name, sheet_name=\"p-v \" + variant)      \n",
    "        vp_df = pd.read_excel(file_name, sheet_name=\"v-p \" + variant)\n",
    "\n",
    "        pv_vcf = parent + \"_\" + variant + \"_p-v.vcf\"\n",
    "        vp_vcf = parent + \"_\" + variant + \"_v-p.vcf\"\n",
    "        \n",
    "        with open(pv_vcf, 'w') as vcf:\n",
    "            vcf.write(header)\n",
    "        \n",
    "        with open(vp_vcf, 'w') as vcf:\n",
    "            vcf.write(header)\n",
    "        \n",
    "        pv_df.to_csv(pv_vcf, sep=\"\\t\", mode='a', index=False)\n",
    "        vp_df.to_csv(vp_vcf, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f136f48c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-09-17T06:52:03.101703Z",
     "start_time": "2021-09-17T06:52:02.807709Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 14</th>\n",
       "      <th>Unnamed: 15</th>\n",
       "      <th>Unnamed: 16</th>\n",
       "      <th>Unnamed: 17</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "      <th>Unnamed: 19</th>\n",
       "      <th>Unnamed: 20</th>\n",
       "      <th>Unnamed: 21</th>\n",
       "      <th>Unnamed: 22</th>\n",
       "      <th>Unnamed: 23</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sample ID</td>\n",
       "      <td>Chr</td>\n",
       "      <td>Start</td>\n",
       "      <td>End</td>\n",
       "      <td>Ref</td>\n",
       "      <td>Alt</td>\n",
       "      <td>Func.refGene</td>\n",
       "      <td>Gene.refGene</td>\n",
       "      <td>GeneDetail.refGene</td>\n",
       "      <td>ExonicFunc.refGene</td>\n",
       "      <td>...</td>\n",
       "      <td>Otherinfo4</td>\n",
       "      <td>Otherinfo5</td>\n",
       "      <td>Otherinfo6</td>\n",
       "      <td>Otherinfo7</td>\n",
       "      <td>Otherinfo8</td>\n",
       "      <td>Otherinfo9</td>\n",
       "      <td>Otherinfo10</td>\n",
       "      <td>Otherinfo11</td>\n",
       "      <td>Otherinfo12</td>\n",
       "      <td>Otherinfo13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6249979</td>\n",
       "      <td>6249979</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR153</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6249979</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>103</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=44;VD=7;AF=0.1591;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>0/1:44:7:37,7:0.1591:20,17:3,4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chr1</td>\n",
       "      <td>12879834</td>\n",
       "      <td>12879834</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PRAMEF4</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>12879834</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>289</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=670;VD=233;AF=0.34...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>0/1:670:233:437,233:0.3478:184,253:114,119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chr1</td>\n",
       "      <td>12893437</td>\n",
       "      <td>12893437</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PRAMEF10</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>12893437</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>267</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=482;VD=166;AF=0.34...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>0/1:482:166:169,166:0.3444:39,130:48,118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chrX</td>\n",
       "      <td>50607778</td>\n",
       "      <td>50607778</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SHROOM4</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>50607778</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>212</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=195;VD=57;AF=0.292...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>0/1:195:57:138,57:0.2923:108,30:45,12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>482</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chrX</td>\n",
       "      <td>136877960</td>\n",
       "      <td>136877960</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>RBMX</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>136877960</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>222</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=136;VD=97;AF=0.713...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>1/0:136:97:37,97:0.7132:17,20:21,76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chrX</td>\n",
       "      <td>136878071</td>\n",
       "      <td>136878071</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>RBMX</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>136878071</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>171</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=40;VD=29;AF=0.725;...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>1/0:40:29:11,29:0.725:0,11:9,20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chrX</td>\n",
       "      <td>136879401</td>\n",
       "      <td>136879401</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>RBMX</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>136879401</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>241</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=SNV;DP=246;VD=105;AF=0.42...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>0/1:246:105:141,105:0.4268:82,59:45,60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>VE_22241</td>\n",
       "      <td>chrX</td>\n",
       "      <td>141906299</td>\n",
       "      <td>141906301</td>\n",
       "      <td>CCC</td>\n",
       "      <td>GCA</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEC1</td>\n",
       "      <td>.</td>\n",
       "      <td>nonframeshift substitution</td>\n",
       "      <td>...</td>\n",
       "      <td>chrX</td>\n",
       "      <td>141906299</td>\n",
       "      <td>.</td>\n",
       "      <td>CCC</td>\n",
       "      <td>GCA</td>\n",
       "      <td>218</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMPLE=VE_22241;TYPE=Complex;DP=459;VD=81;AF=0...</td>\n",
       "      <td>GT:DP:VD:AD:AF:RD:ALD</td>\n",
       "      <td>0/1:459:81:366,81:0.1765:223,143:31,50</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>486 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             1 Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0            2        NaN        NaN        NaN        NaN        NaN   \n",
       "1    Sample ID        Chr      Start        End        Ref        Alt   \n",
       "2     VE_22241       chr1    6249979    6249979          G          T   \n",
       "3     VE_22241       chr1   12879834   12879834          T          G   \n",
       "4     VE_22241       chr1   12893437   12893437          C          G   \n",
       "..         ...        ...        ...        ...        ...        ...   \n",
       "481   VE_22241       chrX   50607778   50607778          T          G   \n",
       "482   VE_22241       chrX  136877960  136877960          C          T   \n",
       "483   VE_22241       chrX  136878071  136878071          C          T   \n",
       "484   VE_22241       chrX  136879401  136879401          C          G   \n",
       "485   VE_22241       chrX  141906299  141906301        CCC        GCA   \n",
       "\n",
       "       Unnamed: 6    Unnamed: 7          Unnamed: 8  \\\n",
       "0             NaN           NaN                 NaN   \n",
       "1    Func.refGene  Gene.refGene  GeneDetail.refGene   \n",
       "2          exonic        GPR153                   .   \n",
       "3          exonic       PRAMEF4                   .   \n",
       "4          exonic      PRAMEF10                   .   \n",
       "..            ...           ...                 ...   \n",
       "481        exonic       SHROOM4                   .   \n",
       "482        exonic          RBMX                   .   \n",
       "483        exonic          RBMX                   .   \n",
       "484        exonic          RBMX                   .   \n",
       "485        exonic        MAGEC1                   .   \n",
       "\n",
       "                     Unnamed: 9  ... Unnamed: 14 Unnamed: 15 Unnamed: 16  \\\n",
       "0                           NaN  ...         NaN         NaN         NaN   \n",
       "1            ExonicFunc.refGene  ...  Otherinfo4  Otherinfo5  Otherinfo6   \n",
       "2             nonsynonymous SNV  ...        chr1     6249979           .   \n",
       "3             nonsynonymous SNV  ...        chr1    12879834           .   \n",
       "4             nonsynonymous SNV  ...        chr1    12893437           .   \n",
       "..                          ...  ...         ...         ...         ...   \n",
       "481           nonsynonymous SNV  ...        chrX    50607778           .   \n",
       "482           nonsynonymous SNV  ...        chrX   136877960           .   \n",
       "483           nonsynonymous SNV  ...        chrX   136878071           .   \n",
       "484           nonsynonymous SNV  ...        chrX   136879401           .   \n",
       "485  nonframeshift substitution  ...        chrX   141906299           .   \n",
       "\n",
       "    Unnamed: 17 Unnamed: 18 Unnamed: 19  Unnamed: 20  \\\n",
       "0           NaN         NaN         NaN          NaN   \n",
       "1    Otherinfo7  Otherinfo8  Otherinfo9  Otherinfo10   \n",
       "2             G           T         103         PASS   \n",
       "3             T           G         289         PASS   \n",
       "4             C           G         267         PASS   \n",
       "..          ...         ...         ...          ...   \n",
       "481           T           G         212         PASS   \n",
       "482           C           T         222         PASS   \n",
       "483           C           T         171         PASS   \n",
       "484           C           G         241         PASS   \n",
       "485         CCC         GCA         218         PASS   \n",
       "\n",
       "                                           Unnamed: 21            Unnamed: 22  \\\n",
       "0                                                  NaN                    NaN   \n",
       "1                                          Otherinfo11            Otherinfo12   \n",
       "2    SAMPLE=VE_22241;TYPE=SNV;DP=44;VD=7;AF=0.1591;...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "3    SAMPLE=VE_22241;TYPE=SNV;DP=670;VD=233;AF=0.34...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "4    SAMPLE=VE_22241;TYPE=SNV;DP=482;VD=166;AF=0.34...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "..                                                 ...                    ...   \n",
       "481  SAMPLE=VE_22241;TYPE=SNV;DP=195;VD=57;AF=0.292...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "482  SAMPLE=VE_22241;TYPE=SNV;DP=136;VD=97;AF=0.713...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "483  SAMPLE=VE_22241;TYPE=SNV;DP=40;VD=29;AF=0.725;...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "484  SAMPLE=VE_22241;TYPE=SNV;DP=246;VD=105;AF=0.42...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "485  SAMPLE=VE_22241;TYPE=Complex;DP=459;VD=81;AF=0...  GT:DP:VD:AD:AF:RD:ALD   \n",
       "\n",
       "                                    Unnamed: 23  \n",
       "0                                           NaN  \n",
       "1                                   Otherinfo13  \n",
       "2                0/1:44:7:37,7:0.1591:20,17:3,4  \n",
       "3    0/1:670:233:437,233:0.3478:184,253:114,119  \n",
       "4      0/1:482:166:169,166:0.3444:39,130:48,118  \n",
       "..                                          ...  \n",
       "481       0/1:195:57:138,57:0.2923:108,30:45,12  \n",
       "482         1/0:136:97:37,97:0.7132:17,20:21,76  \n",
       "483             1/0:40:29:11,29:0.725:0,11:9,20  \n",
       "484      0/1:246:105:141,105:0.4268:82,59:45,60  \n",
       "485      0/1:459:81:366,81:0.1765:223,143:31,50  \n",
       "\n",
       "[486 rows x 24 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(\"results/VE_22241_VE_22334 - Copy.xlsx\")\n",
    "display(df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
